src/core/basics.ml

```diff
@@ -34,19 +34,19 @@ let to_tvar : term -> tvar = fun t ->
     “marshaled” (e.g., by the {!module:Sign} module), as this would break the
     freshness invariant of new variables. *)
 
-(** [sensible_tref t] transforms {!constructor:Appl} into references. *)
-let appl_to_tref : term -> term = fun t ->
-  match t with
-  | Appl(_,_) as t -> TRef(ref (Some t))
-  | t              -> t
-
 (** [count_products a] returns the number of consecutive products at the  head
     of the term [a]. *)
 let rec count_products : term -> int = fun t ->
   match unfold t with
   | Prod(_,b) -> 1 + count_products (Bindlib.subst b Kind)
   | _         -> 0
 
+(** [imply t u] creates the term [t ⇒ u], that is, the dependent product
+    [∀x: t, u] where [u] does not depend on [x]. *)
+let imply : term -> term -> term = fun t u ->
+  let dummy = Bindlib.new_var mkfree "_" in
```

Comment:
`Bindlib.new_var mkfree "_"` occurs several times in the code. Do you think that we should define this constant dummy once and for all in terms.ml?

Reply:
I don't know exactly how Bindlib works, but wouldn't that introduce (undesired) sharing?

Reply:
@rlepigre what do you think?

Reply:
That would not be a problem as long as we do not start using this variable in random places (i.e., it should never appear in terms). We could enforce that by defining a wrapper around `Bindlib.bind_var` as follows, but I would not even bother doing that because the `imply` function is definitely not critical for efficiency.
```ocaml
let bind_dummy_var : term box -> ttbinder box =
  Bindlib.(bind_var (new_var mkfree "_"))
```
You can see that I used a similar scheme in `Terms._Impl``.

Reply:
I like this bind_dummy_var.

Reply:
I'd be in favour of providing such a function as it would prevent people that discover the code to do such a mistake.

Perhaps there should be a more thorough _contribution guide_ that indicates which function to consider to do X, which comments to read particularly &c.

Reply:
On the other hand, we often need the dummy variable, like in, for instance, `extend_meta_type`, so it might not be that relevant.

Reply:
I really think that function would be useless.

Reply:
ok, let's remove it if it is used nowhere for the moment (it is easy to add it again if needed)